source/device.hpp

```diff
@@ -29,7 +29,10 @@ class IRoutingDevice {
     virtual void add_inlink(std::shared_ptr<Link> link) = 0;
     virtual void update_routing_table(std::shared_ptr<IRoutingDevice> dest,
                                       std::shared_ptr<Link> link) = 0;
+
     virtual std::shared_ptr<Link> next_inlink() = 0;
+    virtual std::shared_ptr<Link> get_destination(
```

Comment:
I think get_link_to_destination would be more clear:
```suggestion
    virtual std::shared_ptr<Link> get_link_to_destination(
```

Reply:
I think you are right. I will change `IRoutingDevice` interface